src/number_theory/bernoulli.lean

```diff
@@ -258,3 +258,122 @@ begin
   { ext x, rw bernoulli_eq_bernoulli' (succ_ne_zero x ∘ succ.inj) },
   { ring },
 end
+
+theorem bernoulli_power_series :
+  (exp ℚ - 1) * mk (λ n, (bernoulli n / n!)) = X :=
+begin
+  have h: power_series.mk (λ n, (bernoulli' n / n! : ℚ)) * (exp ℚ - 1)
+    = X * exp ℚ :=
+  begin
+    simp only [bernoulli'_power_series],
+  end,
+  sorry,
+end
+
+section faulhaber
+
+lemma aux_cauchy_prod (n : ℕ) :
+mk (λ (p : ℕ), bernoulli p / ↑p!) *
+ mk (λ (q : ℕ), (coeff ℚ (q + 1)) (exp ℚ ^ n))
+ = mk (λ (p : ℕ), (finset.range (p + 1)).sum (λ (i : ℕ),
+ bernoulli i * ↑((p + 1).choose i) * ↑n ^ (p + 1 - i) / ↑(p + 1)!)) :=
+begin
+    ext q,
+  rw [power_series.coeff_mul],
+  simp only [coeff_mk, coeff_mk, nat.cast_mul],
+  let f : ℕ → ℕ → ℚ := λ (a : ℕ), λ (b : ℕ),
+  bernoulli a / ↑a! * (coeff ℚ (b + 1)) (exp ℚ ^ n),
+  rw [finset.nat.sum_antidiagonal_eq_sum_range_succ f],
+  refine finset.sum_congr _ _,
+  { refl, },
```

Comment:
```suggestion
  refine finset.sum_congr rfl _,
```
and then the indentation can be reduced below

Reply:
fixed